src/main/java/net/minecraftforge/event/entity/living/LootingLevelEvent.java

```diff
@@ -34,6 +36,7 @@ public LootingLevelEvent(LivingEntity entity, DamageSource damageSource, int loo
         this.lootingLevel = lootingLevel;
     }
 
+    @Nullable
     public DamageSource getDamageSource() {
```

Comment:
This 1) 1/2 does a @Nullable you'd need it on the field, and constructor argument
and 2) Doesn't make any sense. Things shouldn't be null. If people pass in null and cause NPEs in consumers that's the invokers issue.

Reply:
I agree with you that it would have been nicer if the damage source was never null. The original reason for why I created this pull request was that several people complained that my mod was crashing, when in fact it was other mods causing NPE because of getDamageSource() being null. I'll add the nullable annotation all the way up the chain then!